File: push-to-github.js
#!/usr/bin/env node

import { execSync } from 'child_process';
import fs from 'fs';
import path from 'path';

// Get environment variables
const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
const GITHUB_REPO = process.env.GITHUB_REPO;
const GITHUB_OWNER = process.env.GITHUB_OWNER;

// Validate environment variables
if (!GITHUB_TOKEN || !GITHUB_REPO || !GITHUB_OWNER) {
  console.error('❌ Missing required environment variables:');
  console.error('   GITHUB_TOKEN:', !!GITHUB_TOKEN ? '✅ Set' : '❌ Missing');
  console.error('   GITHUB_REPO:', !!GITHUB_REPO ? '✅ Set' : '❌ Missing'); 
  console.error('   GITHUB_OWNER:', !!GITHUB_OWNER ? '✅ Set' : '❌ Missing');
  process.exit(1);
}

console.log('🚀 Starting GitHub push process...');
console.log(`📂 Repository: ${GITHUB_OWNER}/${GITHUB_REPO}`);

function runCommand(command, description) {
  try {
    console.log(`\n📋 ${description}...`);
    const result = execSync(command, { 
      stdio: 'pipe', 
      encoding: 'utf-8',
      cwd: process.cwd()
    });
    console.log(`✅ ${description} completed`);
    return result;
  } catch (error) {
    console.error(`❌ ${description} failed:`);
    console.error(error.message);
    if (error.stdout) console.error('STDOUT:', error.stdout);
    if (error.stderr) console.error('STDERR:', error.stderr);
    throw error;
  }
}

async function pushToGitHub() {
  try {
    // Check if git is initialized
    const isGitRepo = fs.existsSync('.git');
    
    if (!isGitRepo) {
      runCommand('git init', 'Initializing Git repository');
      runCommand('git branch -M main', 'Setting main branch');
    } else {
      console.log('✅ Git repository already initialized');
    }

    // Configure git user (for commits)
    try {
      runCommand('git config user.email "action@github.com"', 'Setting Git email');
      runCommand('git config user.name "AJAI Bot"', 'Setting Git username');
    } catch (error) {
      console.log('ℹ️ Git config already set or failed to set');
    }

    // Add remote origin with token authentication
    const remoteUrl = `https://${GITHUB_TOKEN}@github.com/${GITHUB_OWNER}/${GITHUB_REPO}.git`;
    
    try {
      runCommand('git remote remove origin', 'Removing existing origin');
    } catch (error) {
      console.log('ℹ️ No existing origin to remove');
    }
    
    runCommand(`git remote add origin ${remoteUrl}`, 'Adding GitHub remote');

    // Add all files (respecting .gitignore)
    runCommand('git add .', 'Adding all files');

    // Check if there are any changes to commit
    try {
      const status = runCommand('git status --porcelain', 'Checking Git status');
      if (!status.trim()) {
        console.log('ℹ️ No changes to commit');
        return;
      }
    } catch (error) {
      // Continue if status check fails
    }

    // Commit changes
    const commitMessage = `🚀 Update AJAI Trading Platform - ${new Date().toISOString()}

🔧 Features:
- AI-powered trading signals with 85%+ confidence auto-execution
- Real-time market data analysis and portfolio management
- News sentiment analysis and market impact predictions
- Advanced risk management and position tracking
- Interactive chat assistant (AJxAI) with trading insights
- Comprehensive dashboard with live market indicators

💻 Tech Stack:
- Frontend: React + TypeScript + Tailwind CSS
- Backend: Node.js + Express + PostgreSQL
- AI: OpenAI API integration for market analysis
- Real-time: WebSocket connections for live updates
- Database: PostgreSQL with Drizzle ORM`;

    runCommand(`git commit -m "${commitMessage}"`, 'Committing changes');

    // Push to GitHub
    runCommand('git push -u origin main', 'Pushing to GitHub');

    console.log('\n🎉 SUCCESS! Your AJAI Trading Platform has been pushed to GitHub!');
    console.log(`🔗 Repository: https://github.com/${GITHUB_OWNER}/${GITHUB_REPO}`);
    console.log(`📊 View your code: https://github.com/${GITHUB_OWNER}/${GITHUB_REPO}/tree/main`);

  } catch (error) {
    console.error('\n❌ Push to GitHub failed:', error.message);
    process.exit(1);
  }
}

// Run the push process
pushToGitHub();

File: push_to_github.py
#!/usr/bin/env python3

import os
import subprocess
import sys
from datetime import datetime

def run_command(command, description, cwd=None):
    """Run a command and handle errors"""
    print(f"📋 {description}...")
    try:
        result = subprocess.run(
            command, 
            shell=True, 
            check=True, 
            capture_output=True, 
            text=True,
            cwd=cwd or os.getcwd()
        )
        print(f"✅ {description} completed")
        return result.stdout.strip()
    except subprocess.CalledProcessError as e:
        print(f"❌ {description} failed:")
        print(f"Error: {e.stderr}")
        if e.stdout:
            print(f"Output: {e.stdout}")
        return None

def check_git_repo():
    """Check if git repository exists"""
    return os.path.exists('.git')

def push_to_github():
    """Main function to push code to GitHub"""
    
    print("🚀 AJAI Trading Platform - GitHub Push Script")
    print("=" * 50)
    
    # Get environment variables
    github_token = os.getenv('GITHUB_TOKEN')
    github_repo = os.getenv('GITHUB_REPO') 
    github_owner = os.getenv('GITHUB_OWNER')
    
    # Validate environment variables
    if not all([github_token, github_repo, github_owner]):
        print("❌ Missing required environment variables:")
        print(f"   GITHUB_TOKEN: {'✅ Set' if github_token else '❌ Missing'}")
        print(f"   GITHUB_REPO: {'✅ Set' if github_repo else '❌ Missing'}")
        print(f"   GITHUB_OWNER: {'✅ Set' if github_owner else '❌ Missing'}")
        print("\nPlease set these in your Replit Secrets!")
        return False
    
    print(f"📂 Repository: {github_owner}/{github_repo}")
    print()
    
    # Check if git is initialized
    if not check_git_repo():
        if run_command("git init", "Initializing Git repository") is None:
            return False
        if run_command("git branch -M main", "Setting main branch") is None:
            return False
    else:
        print("✅ Git repository already initialized")
        print()
    
    # Configure git user
    if run_command('git config user.email "ajai-bot@example.com"', "Setting Git email") is None:
        print("⚠️ Could not set git email, continuing...")
    
    if run_command('git config user.name "AJAI Bot"', "Setting Git username") is None:
        print("⚠️ Could not set git username, continuing...")
    
    # Remove existing origin if it exists
    subprocess.run("git remote remove origin", shell=True, capture_output=True)
    print("ℹ️ Removed existing origin (if any)")
    
    # Add remote origin with token authentication
    remote_url = f"https://{github_token}@github.com/{github_owner}/{github_repo}.git"
    if run_command(f'git remote add origin {remote_url}', "Adding GitHub remote") is None:
        return False
    
    # Add all files
    if run_command("git add .", "Adding all files") is None:
        return False
    
    # Check if there are changes to commit
    try:
        result = subprocess.run("git diff-index --quiet HEAD --", shell=True, check=True)
        print("ℹ️ No changes to commit")
        return True
    except subprocess.CalledProcessError:
        # There are changes to commit
        pass
    
    # Create detailed commit message
    timestamp = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S')
    commit_message = f"""🚀 AJAI Trading Platform Update - {timestamp}

🔧 Features:
- AI-powered trading signals with 85%+ confidence auto-execution
- Real-time market data analysis and portfolio management  
- News sentiment analysis and market impact predictions
- Advanced risk management and position tracking
- Interactive chat assistant (AJxAI) with trading insights
- Comprehensive dashboard with live market indicators
- Unified navigation system with seamless page routing

💻 Tech Stack:
- Frontend: React + TypeScript + Tailwind CSS
- Backend: Node.js + Express + PostgreSQL
- AI: OpenAI API integration for market analysis
- Real-time: WebSocket connections for live updates
- Database: PostgreSQL with Drizzle ORM
- UI: shadcn/ui + Radix UI components

🚀 Ready for deployment and live trading!"""
    
    # Commit changes
    if run_command(f'git commit -m "{commit_message}"', "Committing changes") is None:
        return False
    
    # Push to GitHub
    if run_command("git push -u origin main", "Pushing to GitHub") is None:
        print("⚠️ Push failed, trying force push...")
        if run_command("git push -f origin main", "Force pushing to GitHub") is None:
            return False
    
    print()
    print("🎉 SUCCESS! Your AJAI Trading Platform has been pushed to GitHub!")
    print(f"🔗 Repository: https://github.com/{github_owner}/{github_repo}")
    print(f"📊 View your code: https://github.com/{github_owner}/{github_repo}/tree/main")
    print()
    print("🚀 Your AI trading platform is now live on GitHub!")
    
    return True

if __name__ == "__main__":
    try:
        success = push_to_github()
        if not success:
            print("\n❌ Push to GitHub failed!")
            sys.exit(1)
    except KeyboardInterrupt:
        print("\n⛔ Push cancelled by user")
        sys.exit(1)
    except Exception as e:
        print(f"\n❌ Unexpected error: {e}")
        sys.exit(1)